src/com/organizationmanager/OrganizationXMLParser.java

```
@@ -0,0 +1,82 @@
+package com.organizationmanager;
+
+import java.io.File;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.TreeSet;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+public class OrganizationXMLParser {
+    private File xmlFile;
+
+    public OrganizationXMLParser(String fileName) {
+        this.xmlFile = new File(fileName);
+    }
+
+    public TreeSet<Organization> loadOrganizations() {
+        TreeSet<Organization> organizations = new TreeSet<>();
+        try {
+            if (!xmlFile.exists()) {
+                System.err.println("Error: The file " + xmlFile.getName() + " does not exist.");
+                return organizations;
+            }
+
+            DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+            DocumentBuilder builder = factory.newDocumentBuilder();
+            Document document = builder.parse(xmlFile);
+            document.getDocumentElement().normalize();
+
+            NodeList nodeList = document.getElementsByTagName("organization");
+            for (int i = 0; i < nodeList.getLength(); i++) {
+                Node node = nodeList.item(i);
+                if (node.getNodeType() == Node.ELEMENT_NODE) {
+                    Element element = (Element) node;
+                    Organization org = parseOrganization(element);
+                    if (org != null) {
+                        organizations.add(org);
+                    }
+                }
+            }
+        } catch (Exception e) {

```

Don't do catch-all exceptions

---

Why did you ignore my comment?